### PR TITLE
recoll: 1.20.6 -> 1.21.3

### PR DESCRIPTION
**Pull request by bnikolic**
cc @jcumming

### CODE REVIEW

File: pkgs/applications/search/recoll/default.nix [hunk 1]
```diff
@@ -7,17 +7,19 @@
...
   name = "recoll-${ver}";
 
   src = fetchurl {
     url = "http://www.lesbonscomptes.com/recoll/${name}.tar.gz";
-    sha256 = "0ympk2w21cxfvysyx96p0npsa54csfc84cicpi8nsj1qp824zxwq";
+    sha256 = "66f039f08b149d5e4840664d4a636f6b55145b02072f87aab83282ebe0cd593a";
   };
 
+  patches = [ ./nodeblayout.patch ./versionawk.patch ];
+
```
Comment (jcumming) [inline pkgs/applications/search/recoll/default.nix]: This would be a good place for comments or links to the issues the patches fix.

**Comment by jcumming** [comment]
:+1:

File: pkgs/applications/search/recoll/versionawk.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
++++ recoll-1.21.3/filters/rclpdf	2015-12-14 21:37:01.984945925 +0000
+@@ -129,7 +129,7 @@
+ {
+     # Test poppler version: at some point before 0.24, poppler began
+     # to properly escape text inside the header (but not the body).
+-    XYZ=`pdftotext -v 2>&1 | awk '/pdftotext/{print $3}'`
++    XYZ=`pdftotext -v 2>&1 | awk '/version/{print $3}'`
+     MAJOR=`echo $XYZ | cut -d. -f 1`
+     MINOR=`echo $XYZ | cut -d. -f 2`
+     escapeheader=1
```
Comment (joachifm) [inline pkgs/applications/search/recoll/versionawk.patch]: On my machine, this change is pointless: it changes the match clause, but you're still matching the same line in the output from `pdftotext -v`. What problem does this solve?

Comment (bnikolic) [inline pkgs/applications/search/recoll/versionawk.patch]: joachifm <notifications@github.com> writes:

>> @@ -0,0 +1,12 @@
>> +diff -ru recoll-1.21.3-orig/filters/rclpdf recoll-1.21.3/filters/rclpdf
>> +--- recoll-1.21.3-orig/filters/rclpdf	2015-09-28 08:08:15.000000000 +0100
>> ++++ recoll-1.21.3/filters/rclpdf	2015-12-14 21:37:01.984945925 +0000
>> +@@ -129,7 +129,7 @@
>> + {
>> +     # Test poppler version: at some point before 0.24, poppler began
>> +     # to properly escape text inside the header (but not the body).
>> +-    XYZ=`pdftotext -v 2>&1 | awk '/pdftotext/{print $3}'`
>> ++    XYZ=`pdftotext -v 2>&1 | awk '/version/{print $3}'`
>> +     MAJOR=`echo $XYZ | cut -d. -f 1`
>> +     MINOR=`echo $XYZ | cut -d. -f 2`
>> +     escapeheader=1
>
> On my machine, this change is pointless: it changes the match clause, but you're still matching the same line in the output from `pdftotext -v`. What problem does this solve?

It avoids later corruption of this shell script by the

   "substituteInPlace  $f --replace pdftotext   ${poppler_utils}/bin/pdftotext "

postInstall command.

-- 
Bojan Nikolic          ||          http://www.bnikolic.co.uk

**Pull request merged**